### PR TITLE
Refs [Task][42190]  Admin - Show all buyers

### PR DESCRIPTION
**Pull request by TranHuyHoang95**
## Related Tickets
- [#42190](https://edu-redmine.sun-asterisk.vn/issues/42190)

## WHAT
- Write RSpec for Admin::UsersController

## HOW
- Install gems for write RSpec
- Pass 100% coverage for Admin::UsersConntroller
- Use FactoryBot to make database test for RSpec
- Use share_examples to refactor code

## Evidence
image

image

### CODE REVIEW

**Comment by TranHuyHoang95** [comment]
ready

**Comment by vanvtt-0952** [comment]
commented
2 commit

File: app/controllers/admin/orders_controller.rb [hunk 1]
```diff
@@ -25,6 +25,10 @@ def load_orders
     @orders = Order.includes(order_products: :product)
                    .order_by_status.page(params[:page])
                    .per(Settings.per_page.per_5)
+    return if @orders
+
+    flash[:danger] = t "admin.order.nil"
+    redirect_to admin_root_path
```
Comment (vanvtt-0952) [inline app/controllers/admin/orders_controller.rb]: 2 line này có khi nào dc chạy nko em
orders luôn khác nil mà

File: spec/controllers/admin/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          expect(response).to render_template("index")
+        end
+
+      end
+
+    end
+
+  end
+
+end
```
Comment (vanvtt-0952) [inline spec/controllers/admin/users_controller_spec.rb]:
```suggestion
        end
      end
    end
  end
end
```

File: spec/controllers/products_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+RSpec.describe ProductsController, type: :controller	do
+  let!(:category){FactoryBot.create :category}
+  let!(:product){FactoryBot.create :product, category_id: category.id}
+  describe "#show" do
+    context "when product exists" do
+      it "assign product is match with product" do
+        get :show, params: {id: product.id, category_id: category.id}
+        expect(assigns(:product)).to	eq product
+      end
+
```
Comment (vanvtt-0952) [inline spec/controllers/products_controller_spec.rb]:
```suggestion
```

File: app/controllers/admin/orders_controller.rb (hunk 1 above)
Comment (TranHuyHoang95) [reply app/controllers/admin/orders_controller.rb]: Dạ để em check lại chỗ ni

Comment (TranHuyHoang95) [reply app/controllers/admin/orders_controller.rb]: Dạ. Có show flash ở view rồi mà em quơn.

**Comment by TranHuyHoang95** [comment]
ready

**Pull request merged**